src/ParseServer.js

```diff
@@ -171,7 +177,11 @@ class ParseServer {
     const userController = new UserController(emailControllerAdapter, appId, { verifyUserEmails });
     const liveQueryController = new LiveQueryController(liveQuery);
 
+    if (settingsCacheOptions) {
+      cache.apps = PersistentSettingsStore(settingsCacheOptions, definedOptions);
```

Comment:
Defined settings are modifiable if lockDefinedSettings is false (so they will be overwritten by their persisted values).

Locked settings (like masterKey) are settings where I wasn't sure how to approach modifying them without a restart due to side effects.  For example, if you change the masterKey from the parse dashboard all the sudden your experience is broken as you can't connect to the server anymore.